Sources_v3/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift

```diff
@@ -0,0 +1,222 @@
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+    
+    // MARK: - Init
+    
+    public required init(
+        channel: _ChatChannel<ExtraData>? = nil,
+        uiConfig: UIConfig<ExtraData> = .default
+    ) {
+        self.channel = channel
+        self.uiConfig = uiConfig
+        
+        super.init(frame: .zero)
+        
+        commonInit()
+    }
+    
+    public required init?(coder: NSCoder) {
+        self.uiConfig = .default
+        self.channel = nil
+        
+        super.init(coder: coder)
+        
+        commonInit()
+    }
+    
+    public func commonInit() {
+        setupAppearance()
+        setupLayout()
+        updateContent()
+    }
+    
+    // MARK: - Public
+    
+    open func setupAppearance() {
+        titleLabel.numberOfLines = 1
+        titleLabel.font = .boldSystemFont(ofSize: .scaledFontSize(for: 14))
+        
+        subtitleLabel.numberOfLines = 1
+        subtitleLabel.textColor = .gray
+        subtitleLabel.font = .systemFont(ofSize: .scaledFontSize(for: 12.5))
+        
+        timestampLabel.numberOfLines = 1
+        timestampLabel.textColor = .gray
+        timestampLabel.font = .systemFont(ofSize: .scaledFontSize(for: 12),
+                                          weight: .medium)
+    }
+    
+    open func setupLayout() {
+        embed(container)
+        
+        container.leftStackView.isHidden = false
+        container.leftStackView.alignment = .center
+        
+        container.leftStackView.addArrangedSubview(avatarView)
+        
+        container.centerStackView.isHidden = false
+        container.centerStackView.axis = .vertical
+        container.centerStackView.spacing = 2
+        
+        container.centerStackView.addArrangedSubview(titleLabel)
+        container.centerStackView.addArrangedSubview(subtitleLabel)
+        
+        container.rightStackView.isHidden = false
+        container.rightStackView.axis = .vertical
+        container.rightStackView.spacing = 6
+        
+        let rightTopContainer = UIStackView()
+        
+        rightTopContainer.addSpacer()
+        rightTopContainer.addArrangedSubview(unreadCountBubble)
+        
+        let rightBottomContainer = UIStackView()
+        rightBottomContainer.spacing = 6
+        
+        rightBottomContainer.addSpacer()
+        rightBottomContainer.addArrangedSubview(readStatusView)
+        rightBottomContainer.addArrangedSubview(timestampLabel)
+        
+        container.rightStackView.addSpacer()
+        container.rightStackView.addArrangedSubview(rightTopContainer)
+        container.rightStackView.addArrangedSubview(rightBottomContainer)
+    
+        avatarView.widthAnchor.constraint(equalToConstant: 56).isActive = true
+        readStatusView.widthAnchor.constraint(equalToConstant: 16).isActive = true
+        unreadCountBubble.heightAnchor.constraint(equalToConstant: 16).isActive = true
+        
+        container.layoutIfNeeded()
```

Comment:
We invoke `layoutIfNeeded` so that the `layoutSubviews` is triggered and the correct margins are set for the container. Did I get it right? 😄

Reply:
That was obsolete :)